### ISSUE TITLE
Flash Firmware 

### ISSUE THREAD

**Issue by distebia** [phil-barrett/grblHAL-teensy-4.x#84]
Salve, ho caricato il firmware generato da web build, tutto perfetto e velocissimo. in mente persiste una domanda: mettiamo che il firmware caricato ha bisogno di una variazione per adattarsi alla macchina, genero nuovamente il firmware e lo carico ma devo cancellare quello esistente oppure la procedura è automatica? In sostanza devo formattare il teensy 4.1 prima del nuovo flash?

**Comment by phil-barrett** [comment]
No, just upload the newer firmware. It will preserve your previous settings.

**Comment by distebia** [comment]
ma se per qualche motivo devo riportarlo a zero e ripartire? esiste tipo un comando erase_flash?

**Comment by phil-barrett** [comment]
Teensy.exe will erase before programming.

If you want to reset grblHAL, *rst will reset all settings back to default values.

**Comment by distebia** [comment]
Salve, grblhal ha implementato la possibilità di avere + mandrini (anche pwm) attivi allo stesso tempo, o magari attivi uno alla volta...su teensy 4.1 e T41U5XBB quale potrebbe essere un valido pin in grado di erogare un pwm 0-5V?

**Comment by phil-barrett** [comment]
Many of the pins on the teensy are capable of 5V PWM. I don't believe you can use them via the Web Builder, though. You would have to change source code and recompile. In addition,  the T41U5XBB will only support pins set up for output - top and right edges on the board.
